index.bs

```diff
@@ -90,37 +90,54 @@ A <dfn export id=file>file entry</dfn> additionally consists of
 a <dfn for="file entry">lock</dfn> (a string that may exclusively be "`open`", "`taken-exclusive`" or "`taken-shared`")
 and a <dfn for="file entry">shared lock count</dfn> (a number representing the number shared locks that are taken at a given point in time).
 
+A user agent has an associated <dfn>file system lock queue</dfn> which is the
+result of [=starting a new parallel queue=]. This queue is to be used for all
+[=tasks=] involving a [=file entry/lock=].
+
 <div algorithm>
-To <dfn for="file entry/lock">take</dfn> a [=file entry/lock=] with a |value| of "`exclusive`" or "`shared`" on a given [=file entry=] |file|:
-
-1. Let |lock| be the |file|'s [=file entry/lock=].
-1. Let |count| be the |file|'s [=file entry/shared lock count=].
-1. If |value| is "`exclusive`":
-   1. If |lock| is "`open`":
-     1. Set lock to "`taken-exclusive`".
-     1. Return true.
-1. If |value| is "`shared`":
-   1. If |lock| is "`open`":
-     1. Set |lock| to "`taken-shared`".
-     1. Set |count| to 1.
-     1. Return true.
-   1. Otherwise, if |lock| is "`taken-shared`":
-     1. Increase |count| by one.
-     1. Return true.
-1. Return false.
+To <dfn for="file entry/lock">take</dfn> a [=file entry/lock=] with a |value| of
+"`exclusive`" or "`shared`" on a given [=file entry=] |file| and with
+algorithm |resultAlgorithm|:
+
+1. [=Enqueue the following steps=] to the [=file system lock queue=]:
+  1. Let |lock| be the |file|'s [=file entry/lock=].
+  1. Let |count| be the |file|'s [=file entry/shared lock count=].
+  1. If |value| is "`exclusive`":
+     1. If |lock| is "`open`":
+       1. Set lock to "`taken-exclusive`".
+       1. [=Queue a storage task=] given |file|'s [=relevant global object=] and
+          |resultAlgorithm| with a value of "`success`".
+       1. Return.
```

Comment:
We use "Abort these steps" in "in parallel" algorithms to not confuse readers they might actually return something useful to someone.